source/docs/listener.md

```diff
@@ -0,0 +1,26 @@
+# Listener implementation
+
+## Socket handling
+
+Envoy uses the following procedure for creating sockets and assigning them to workers.
+
+1. When a listener is created, a socket is pre-created for every worker on the main thread. This
+   allows most errors to be caught early on in the listener creation process (e.g., bad socket
+   option, unable to bind, etc.).
+    * If using reuse port, a unique socket is created for every worker.
+    * If not using reuse port, a unique socket is created for worker 0, and then that socket
+      is duplicated for all other workers.
+2. In the case of a listener update in which the address/socket is to be shared, all sockets are
```

Comment:
"address/socket is to be shared": what does that mean? Maybe give an example of when that happens.